imagery/i.svm.predict/main.c

```diff
@@ -0,0 +1,419 @@
+ * PURPOSE:      Predicts values with Support Vector Machine classifier
+ *
+ * COPYRIGHT:    (C) 2021 by Maris Nartiss and the GRASS Development Team
+ *
+ *               This program is free software under the GNU General Public
+ *               License (>=v2). Read the file COPYING that comes with GRASS
+ *               for details.
+ *
+ *               Development of this module was supported from
+ *               science funding of University of Latvia (2020/2021).
+ * 
+ *****************************************************************************/
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include <unistd.h>
+
+#include <libsvm/svm.h>
+
+#include <grass/gis.h>
+#include <grass/raster.h>
+#include <grass/imagery.h>
+#include <grass/glocale.h>
+
+#define XSTR(s) STR(s)
+#define STR(s) #s
+
+
+/* LIBSVM message wrapper */
+void print_func(const char *s)
+{
+    G_verbose_message("%s", s);
+};
+
+
+int main(int argc, char *argv[])
+{
+    struct GModule *module;
+    struct Option *opt_group, *opt_subgroup, *opt_sigfile, *opt_values;
+    struct Option *opt_svm_cache_size;
+
+    char name_values[GNAME_MAX], name_sigfile[GNAME_MAX];
+    char name_group[GNAME_MAX], name_subgroup[GNAME_MAX];
+    char mapset_values[GMAPSET_MAX], mapset_sigfile[GMAPSET_MAX];
+    char mapset_group[GMAPSET_MAX], mapset_subgroup[GMAPSET_MAX];
+    char sigfile_dir[GPATH_MAX], model_file[GPATH_MAX];
+
+    struct Ref group_ref;
+    char **names_ordered, **mapsets_ordered;
+    const char **semantic_labels_group, **semantic_labels_svm;
+    char semantic_label[GNAME_MAX];
+    int semantic_label_count = 0, semantic_label_match_count = 0;
+
+    struct svm_model *model;
+
+    struct History history;
+    FILE *misc_file;
+    int sigfile_version;
+    double *rescale, r;
+    int rescale_count = 0;
+
+    G_gisinit(argv[0]);
+
+    module = G_define_module();
+    G_add_keyword(_("imagery"));
+    G_add_keyword(_("svm"));
+    G_add_keyword(_("classification"));
+    G_add_keyword(_("prediction"));
+    G_add_keyword(_("regression"));
+    module->description = _("Predict with a SVM");
+
+    opt_group = G_define_standard_option(G_OPT_I_GROUP);
+    /* GTC: SVM prediction input */
+    opt_group->description = _("Maps with feature values (attributes)");
+
+    opt_subgroup = G_define_standard_option(G_OPT_I_SUBGROUP);
+    opt_subgroup->required = NO;
+
+    opt_sigfile = G_define_option();
+    opt_sigfile->key = "signaturefile";
+    opt_sigfile->type = TYPE_STRING;
+    opt_sigfile->key_desc = "name";
+    opt_sigfile->required = YES;
+    opt_sigfile->gisprompt = "old,signatures/libsvm,sigfile";
+    opt_sigfile->description = _("Name of input file containing signatures");
+
+    opt_values = G_define_standard_option(G_OPT_R_OUTPUT);
+    opt_values->required = YES;
+    opt_values->description =
+        _("Output map with predicted class / calculated value");
+
+    opt_svm_cache_size = G_define_option();
```

Comment:
No. It is not. It is only the cache size and only is a part of overall memory consumption.

Reply:
This should be mentioned in the description because user should be informed about that. See #3070 which fixes that for cases where the base consumption is potentially large. If it is not large, than it does not matter and G_OPT_MEMORYMB can be used right away.

Reply:
I added an explanation to the documentation and also improved the description of parameter.